shell/packages/sandstorm-db/db.js

```diff
@@ -1454,6 +1454,23 @@ if (Meteor.isServer) {
     });
   };
 
+  SandstormDb.prototype.deleteUnusedAccount = function(identityId) {
+    // If there is an *unused* account that has `identityId` as a login identity, deletes it.
+
+    check(identityId, String);
+    var account = Meteor.users.findOne({"loginIdentities.id": identityId});
+    if (account &&
+        account.loginIdentities.length == 1 &&
+        account.nonloginIdentities.length == 0 &&
+        !Grains.findOne({userId: account._id}) &&
+        !ApiTokens.findOne({accountId: account._id}) &&
+        (!account.plan || account.plan === "free") &&
```

Comment:
Also check that account.payments is not present -- otherwise they might have credit that they're losing.